### DISCUSSION TITLE
Connect esp8266/32 to RAMPS 1.6

### DISCUSSION THREAD

**Discussion by ducntq** [luc-github/ESP3D#932] [Questions]
Hi

I'm new to 3D printing, and I've been researching about building my own 3D printer. I'm wondering whether I can connect an esp8266/32 directly to a RAMPS 1.6 without the need of using an Arduino Mega?
Anothet stupid question is if this is possible, where will I flash the Marlin firmware?

**Comment by luc-github** [comment]
Marlin FW support ESP32 as main board so yes
ESP8266 is not powerful enough for handling Marlin

**Reply by ducntq** [reply]
Thanks for the information. I have several esp8266/32 lying around, so I don't want to buy an arduino mega.

**Reply by Onabitec** [reply]
Pls am very new to 3d printer ... 
Am trying to build my own 3d printer using few electronics knowledge I have .
And have built it but I have problem with it ... I have calibrate my x and y axis but the z axis have been giving me problem.

If I calibrate my z axis now it will work fine ... And z will move slow 
Later the z axis will just started to be moving fast and it wont be accurate again 

Is like is change the setting by itself .

Pls help
